### PR TITLE
[Parse] Don't skip '}' when recovering from 'default' outside 'switch'

### PR DESCRIPTION
**Pull request by rintaro**
That caused assertion failures in code completion.

rdar://78781163

### CODE REVIEW

**Comment by rintaro** [comment]
@swift-ci Please smoke test

File: lib/Parse/ParseStmt.cpp [hunk 1]
```diff
@@ -517,7 +517,7 @@ static ParserResult<Stmt> recoverFromInvalidCase(Parser &P) {
   assert(P.Tok.is(tok::kw_case) || P.Tok.is(tok::kw_default)
```
Comment (ahoppen) [inline lib/Parse/ParseStmt.cpp:517]: I think you also need to adjust the doc comment `consuming up to the next ':' or '}'`

**Review by ahoppen — APPROVED** [review]
Looks reasonable to me.

**Comment by rintaro** [comment]
@swift-ci Please smoke test

**Comment by rintaro** [comment]
@swift-ci Please smoke test macOS

**Pull request merged**